### PR TITLE
test: ensure that the scale-down behaviour remain consistent

### LINKED ISSUE #1783: add an unit test to ensure that the scale-down behaviour remain consistent
We should implement a unit test to verify that a CNPG cluster when interrupted during the scaling-up process (e.g., by adjusting the instance number), can successfully complete the instance creation process before transitioning to the scaling-down procedure.

### PR DESCRIPTION
**Pull request by armru**
Closes #1783

### CODE REVIEW

**Review by litaocdl — APPROVED** [review]

**Review by litaocdl — APPROVED** [review]

File: controllers/cluster_create.go [hunk 1]
```diff
@@ -1103,8 +1103,9 @@ func (r *ClusterReconciler) ensureInstancesAreCreated(
 		return ctrl.Result{}, nil
 	}
 
+	status := instancesStatus.InstancesReportingStatus()
 	if !cluster.IsNodeMaintenanceWindowInProgress() &&
-		instancesStatus.InstancesReportingStatus() != cluster.Status.ReadyInstances {
+		status != cluster.Status.ReadyInstances {
```
Comment (mnencia) [inline controllers/cluster_create.go:1108]: Use a better name or revert this change, please.

**Review by mnencia — APPROVED** [review]
